roles/kubernetes-apps/external_cloud_controller/openstack/templates/external-openstack-cloud-config.j2

```diff
@@ -45,8 +44,12 @@ manage-security-groups={{ external_openstack_lbaas_manage_security_groups }}
 {% if external_openstack_lbaas_internal_lb is defined %}
 internal-lb={{ external_openstack_lbaas_internal_lb }}
 {% endif %}
-{% if external_openstack_lbaas_use_octavia is defined and external_openstack_lbaas_use_octavia %}
+{% if external_openstack_lbaas_provider is defined %}
+lb-provider={{ external_openstack_lbaas_provider }}
+use-octavia={{ external_openstack_lbaas_provider | lower == 'octavia' }}
+{% else %}
```

Comment:
I think there was a miss here, if "external_openstack_lbaas_provider" isn't defined Octavia activated.

I would suggest replacing this block with something that doesn't change the defaults.
example:
```{% if external_openstack_lbaas_provider is defined %}
lb-provider={{ external_openstack_lbaas_provider }}
use-octavia="{{ 'true' if external_openstack_lbaas_provider == 'octavia' else 'false' }}"
{% endif %}

Reply:
I agree, that would be wiser to keep the default as it is, but [openstack-cloud-controller-manager uses Octavia by default](https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/using-openstack-cloud-controller-manager.md#load-balancer). Kubespray's doc about [openstack external cloud provider](https://github.com/kubernetes-sigs/kubespray/blob/master/docs/openstack.md#the-external-cloud-provider) mentions this as well.

Any other opinion on this ?

Reply:
How can we make some progress here ?

There could be another way of managing this : 
Openstack cloud controller manager (occm) accepts that `use-octavia` and `lb-provider` are set, even if they are conflicting (actually I didn't have a look at the code to see what's the logic in that case).
We could just simply generate the config file from the settings that are defined in kubespray's `openstack.yml` and let occm deals with the values.

What do you think ?